src/lib/lightning/eclair/eclairApi.ts

```diff
@@ -40,3 +42,53 @@ export const httpPost = async <T>(
 ): Promise<T> => {
   return request<T>(node, 'POST', path, body);
 };
+
+const setupConfig = (eclairNode: EclairNode): ELN.ConfigOptions => {
+  const base64auth = Buffer.from(`:${eclairCredentials.pass}`).toString('base64');
+  const config = {
+    url: `127.0.0.1:${eclairNode.ports.rest}`,
+    headers: {
+      Authorization: `Basic ${base64auth}`,
+    },
+  };
+  return config;
+};
```

Comment:
THis is great! Can we also use this in the `request` function to reduce duplication?

Reply:
You can at least reduce the duplication of the base `url` and auth header with the following in the `request` function:
```ts
  const config = setupConfig(node as EclairNode);
  const args = {
    url: `${config.url}/${path}`,
    method,
    body,
    headers: {
      ...config.headers,
      'Content-Type': 'application/x-www-form-urlencoded',
    },
  };
```

You'll just need to include `http://` in the `url` here in `setupConfig`. This already gets stripped out in `listen` by the `stripProtocol` call.
```ts
  const config = {
    url: `http://127.0.0.1:${eclairNode.ports.rest}`,
```